courses/utils.py

```diff
@@ -25,15 +26,28 @@ def ensure_course_run_grade(user, course_run, edx_grade, should_update=False):
 
     updated = False
     if should_update:
-        grade, created = CourseRunGrade.objects.update_or_create(
-            course_run=course_run, user=user, defaults=grade_properties
-        )
-        updated = not created
+        with transaction.atomic():
+            run_grade, created = CourseRunGrade.objects.select_for_update().get_or_create(
+                course_run=course_run, user=user, defaults=grade_properties
+            )
+
+            if (
+                not created
+                and not run_grade.set_by_admin
+                and not has_equal_properties(course_run_grade=run_grade, edx_grade=edx_grade)
```

Comment:
`has_equal_properties(run_grade, edx_grade)` because `has_equal_properties` does not have any kwargs for `course_run_grade` and `edx_grade`. Refer to the later comment regarding this method.